### PR TITLE
Validate input from plugins allocator and detector.

### PR DESCRIPTION
**Pull request by Creatone**

### CODE REVIEW

File: wca/runners/detection.py [hunk 1]
```diff
@@ -33,8 +33,17 @@ def __init__(self):
         self._anomaly_last_occurrence = None
         self._anomaly_counter = 0
 
+    def validate(self, anomalies: List[Anomaly]):
+        assert isinstance(anomalies, list), 'Anomalies should be list type'
+
+        for a in anomalies:
+            assert isinstance(a, Anomaly), 'Wrong anomaly type: {}'.format(type(a))
```
Comment (ppalucki) [inline wca/runners/detection.py]: please become familiar with 'config', assure_type function, then you can just
call assure_type(anomalies, List[Anomaly])

File: wca/runners/allocation.py [hunk 2]
```diff
@@ -293,8 +293,12 @@ def _iterate_body(self,
         new_allocations, anomalies, extra_metrics = self._allocator.allocate(
             platform, tasks_measurements, tasks_resources, tasks_labels,
             current_allocations)
+
```
Comment (felidadae) [inline wca/runners/allocation.py]: remove new line

File: tests/config/test_validate_values.py [hunk 3]
```diff
@@ -70,8 +70,8 @@ class FooEnum(Enum):
     ('127.0.0.1:9876', IpPort(max_size=30)),
     ('127.0.0.1:9876', IpPort)
 ])
-def test_assure_type_good(value, expected_type):
-    _assure_type(value, expected_type)
+def testassure_type_good(value, expected_type):
```
Comment (felidadae) [inline tests/config/test_validate_values.py]: test_assure_type_valid

File: tests/config/test_validate_values.py [hunk 4]
```diff
@@ -101,16 +101,16 @@ def test_assure_type_good(value, expected_type):
     ('127.0.0.1:abc', IpPort(), 'valid port'),
     ('127.0.0.1:9876', IpPort(max_size=3), 'too long'),
 ])
-def test_assure_type_invalid(value, expected_type, expected_exception_msg):
+def testassure_type_invalid(value, expected_type, expected_exception_msg):
```
Comment (felidadae) [inline tests/config/test_validate_values.py]: test_assure

File: tests/config/test_validate_values.py [hunk 5]
```diff
@@ -101,16 +101,16 @@ def test_assure_type_good(value, expected_type):
...
+        assure_type(value, expected_type)
 
 
 @pytest.mark.parametrize('value, expected_type, expected_exception_msg', [
     (1, inspect.Parameter.empty, 'missing type'),
     ([1], Iterable[int], 'generic'),
     (1, Mapping[int, str], 'generic'),
 ])
-def test_assure_type_invalid_weak(value, expected_type, expected_exception_msg):
+def testassure_type_invalid_weak(value, expected_type, expected_exception_msg):
```
Comment (felidadae) [inline tests/config/test_validate_values.py]: test_assure

File: wca/runners/allocation.py [hunk 6]
```diff
@@ -295,6 +295,9 @@ def _iterate_body(self,
             current_allocations)
         allocate_duration = time.time() - allocate_start
 
+        # validate
```
Comment (felidadae) [inline wca/runners/allocation.py]: # Validate.

We use in this file full sentences

File: wca/runners/allocation.py [hunk 7]
```diff
@@ -295,6 +295,9 @@ def _iterate_body(self,
...
+        _validate_allocator_allocate(new_allocations, anomalies, extra_metrics)
```
Comment (felidadae) [inline wca/runners/allocation.py]: I would replace the name:
_validate_allocate_return_vals

**Pull request merged**